python/cudf/dataframe/dataframe.py

```diff
@@ -3179,3 +3179,38 @@ def merge(left, right, *args, **kwargs):
                         merge_doc[idx:]])
 
 register_distributed_serializer(DataFrame)
+
+
+def _align_indices(lhs, rhs):
+    """
+    Internal util to align the indices of two DataFrames. Returns a tuple of
+    the aligned dataframes, or the original arguments if the indices are the
+    same, or if rhs isn't a DataFrame.
+    """
+    lhs_out, rhs_out = lhs, rhs
+    if isinstance(rhs, DataFrame) and not lhs.index.equals(rhs.index):
+        df = lhs.merge(rhs,
+                       sort=True,
+                       how='outer',
+                       left_index=True,
+                       right_index=True,
+                       suffixes=('_x', '_y'))
+        df = df.sort_index()
+        lhs_out = DataFrame()
+        rhs_out = DataFrame()
+        for col in df.columns:
+            if isinstance(col, str):
+                if col.endswith('_x'):
+                    lhs_out[col[:-2]] = df[col]
+                    continue
+                if col.endswith('_y'):
+                    rhs_out[col[:-2]] = df[col]
+                    continue
```

Comment:
Using `continue` here makes it much harder to follow the logic, I think it would be clearer if you just used `if`, `elif`, and `else statements here to accomplish the same.